cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -149,6 +192,50 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, useInternalIP);
   }
 
+  private static void verifyPeering(DataprocConf conf, String network, String systemNetwork,
+                                    String projectId, String systemProjectId, PeeringState state) {
+    switch (state) {
+      case NONE:
+        throw new IllegalStateException(String.format("VPC Peering from network '%s' in project '%s' to network '" +
+                                                        "%s' in project '%s' does not exists. Please establish a " +
+                                                        "peering.",
+                                                      systemNetwork, systemProjectId, network, projectId));
+      case INACTIVE:
+        throw new IllegalStateException(String.format("VPC Peering from network '%s' in project '%s' to network " +
+                                                        "'%s' in project '%s' is in INACTIVE state. Please fix the " +
+                                                        "peering setup and ensure it is in ACTIVE state.",
+                                                      systemNetwork, systemProjectId, network, projectId));
+    }
+  }
+
+  private static PeeringState peeringState(@Nullable String systemNetwork, String systemProjectId,
+                                           Network networkInfo, Compute compute) throws IOException {
+    // systemNetwork can be null when CDAP is not running on GCP for example CDAP running in sandbox environment
+    if (systemNetwork == null) {
+      return PeeringState.NONE;
+    }
+
+    Network systemNetworkInfo = getNetworkInfo(systemProjectId, systemNetwork, compute);
+    // Get self link for the network.
+    String systemNetworkSelfLink = systemNetworkInfo.getSelfLink();
+
+    LOG.info(String.format("Self link for the system network is %s", systemNetworkSelfLink));
+    List<NetworkPeering> peerings = networkInfo.getPeerings();
+    // if the customer does not has a peering established at all the peering list is null
+    if (peerings != null) {
```

Comment:
Invert the check and return early. It makes the code cleaner and easier to reason all the states.

```java
if (peerings == null) {
  return PeeringState.NONE;
}
for (...) {
}
```

Reply:
Done.